docs/src/metadata.md

```diff
@@ -1,155 +1,80 @@
-    ⋮
- 0x33
- 0x00
- 0x21
- 0x00
- 0x00
- 0x00
- 0x02
- 0x74
- 0x61
- 0x67
- 0x00
- 0x07
- 0x00
- 0x00
- 0x00
- 0x73
- 0x79
- 0x6d
- 0x62
- 0x6f
- 0x6c
- 0x00
- 0x02
- 0x6e
- 0x61
- 0x6d
- 0x65
- 0x00
- 0x02
- 0x00
- 0x00
- 0x00
- 0x64
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
- 0x00
-
-julia> using DataFrames
-ERROR: ArgumentError: Package DataFrames not found in current path:
-- Run `import Pkg; Pkg.add("DataFrames")` to install the DataFrames package.
-
-Stacktrace:
- [1] require(::Module, ::Symbol) at ./loading.jl:893
-
-julia> # Specify a non-temp directory as the second argument if you want to reuse this environment across sessions.
-       Pkg.activate(jlso)
- Activating environment at `/var/folders/vz/zx_0gsp9291dhv049t_nx37r0000gn/T/Project.toml`
-
-julia> # Load our object and perhaps inspect some properties about it
-       # Could also choose export it to a more transparent format
-       using DataFrames
-
-julia> describe(jlso[:DataFrame])
-4×8 DataFrame
-│ Row │ variable │ mean     │ min      │ median   │ max      │ nunique │ nmissing │ eltype   │
-│     │ Symbol   │ Union…   │ Any      │ Union…   │ Any      │ Union…  │ Union…   │ DataType │
-├─────┼──────────┼──────────┼──────────┼──────────┼──────────┼─────────┼──────────┼──────────┤
-│ 1   │ a        │ 3.0      │ 1        │ 3.0      │ 5        │         │          │ Int64    │
-│ 2   │ b        │ 0.772432 │ 0.512452 │ 0.863122 │ 0.907903 │         │          │ Float64  │
-│ 3   │ c        │          │ a        │          │ e        │ 5       │ 0        │ Any      │
-│ 4   │ d        │ 0.6      │ 0        │ 1.0      │ 1        │         │          │ Bool     │
+```@example metadata-example
+dir = joinpath(dirname(dirname(pathof(JLSO))), "test", "specimens")
+jlso = read(joinpath(dir, "v4_bson_none.jlso"), JLSOFile)
+jlso[:DataFrame]
+```
+
+Unfortunately, we can't load some object in the current environment, so we might try to load the offending package only to find out it isn't part of our current environment.
+```@example metadata-example
+try using DataFrames catch e @warn e end
 ```
 
-In the future, we may add some tooling to make it easier to view and compare these dictionaries, but it's currently unclear if that should live here or in Pkg.jl.
+Okay, so we don't have DataFrames loaded and it isn't part of our current environment.
+Rather than adding every possible package needed to deserialize the objects in the file, we can use the `Pkg.activate` do-block syntax to:
+
+1. Initial the extact environment needed to deserialize our objects
+2. Load our desired dependencies
+3. Migrate our data to a more appropriate long term format
+
+```@example metadata-example
+using Pkg
+
+# Now we can run our conversion logic in an isolated environment
+mktempdir(pwd()) do d
+    cd(d) do
+        # Modify our Manifest to just use the latest release of JLSO
+        delete!(jlso.manifest, "JLSO")
+
+        Pkg.activate(jlso, d) do
+            @eval Main begin
```

Comment:
As you might expect the `@eval` is necessary to call `using` in the do-block syntax. I've made a note about possibly using a macro below.